go-controller/pkg/ovn/controller/services/utils.go

```diff
@@ -25,3 +31,19 @@ func IsHostEndpoint(endpointIP string) bool {
 	}
 	return true
 }
+
+func getExternalIDsForLoadBalancer(service *v1.Service, netInfo util.NetInfo) map[string]string {
+	nsn := ktypes.NamespacedName{Namespace: service.Namespace, Name: service.Name}
+
+	role := types.NetworkRoleDefault
+	if !netInfo.IsDefault() {
+		role = types.NetworkRolePrimary
+	}
+
+	return map[string]string{
+		types.LoadBalancerOwnerExternalID: nsn.String(),
```

Comment:
current external ids are a bit confusing to me.
We use the same lb.Name as before, which only depends on the service object and doesn't include anything network-specific, and it is a primary client index.
So let's say service's network change (I assume that is what happens when you add or delete primary UDN to the namespace?) then 2 controllers will try to update the same db entry (based on lb name), where old controller will try to clean it up and new controller will try to add it?
Also I didn't get if there is a way to have non-primary network service? Considering we store role in externalID, it should be the case, but I am not sure how it should work.

Reply:
Ah are we allowing to move existing resources from one network to another? I thought it wasn't allowed. 

We're not launching the service controller from secondary networks, at least not yet :)

Reply:
we probably should include the network name as an external ID

Reply:
using proper external IDs would prevent such thing from happening, and we could use primaryID instead of name for client index.
that actually even should not be too difficult to implement, as lb.Name is not a server index.
Also if the network change for a service is not allowed, does it mean a service will always be attached to the network that was primary at the service creation time? Do we enforce this somehow?

Reply:
wait wait, I thought that mine were proper external IDs. Are we talking about db indexes again?

Reply:
hehe yeah, but we can skip that part.
I think we may need to add network name to the lb.Name, and we need to explain what happens on namespace primary network change, and decide if we really need network role external id, considering network role is defined by the network name, and we don't support services for secondary networks.

Reply:
"Also if the network change for a service is not allowed, does it mean a service will always be attached to the network that was primary at the service creation time? Do we enforce this somehow?"

I think it should just transition to the new network if the primary network is created after the service. @ricky-rav lets make sure we test this scenario. Can be done in a follow up.